File: files/prime_test.js
const fs = require('fs');

function isPrime(n) {
    if (n < 2) return false;
    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) return false;
    }
    return true;
}

function findPrimes(limit) {
    const startTime = process.hrtime();
    const primes = [];
    for (let i = 2; i < limit; i++) {
        if (isPrime(i)) primes.push(i);
    }
    const endTime = process.hrtime(startTime);
    return [primes.length, endTime[0] + endTime[1] / 1e9];
}

function runTest() {
// Create array from 1000 to 100000 with step of 1000
    const ranges = Array.from(
        { length: 200 }, 
        (_, i) => (i + 1) * 1000
    );
    
    const results = [];
    
    ranges.forEach(range => {
        const [count, duration] = findPrimes(range);
        results.push([range, count, duration]);
        // Add progress indicator
        console.log(`Completed range: ${range}`);
    });
    
    const csv = 'Range,Primes Found,Duration\n' + 
        results.map(row => row.join(',')).join('\n');
    fs.writeFileSync('nodejs_results.csv', csv);}

runTest();

File: files/prime_test.py
import time
import csv
from datetime import datetime

def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True

def find_primes(limit):
    start_time = time.time()
    primes = [num for num in range(2, limit) if is_prime(num)]
    end_time = time.time()
    return len(primes), end_time - start_time

def run_test():
    results = []
    # Generate ranges from 1000 to 100000 in steps of 1000
    ranges = range(1000, 200001, 1000)
    
    for r in ranges:
        count, duration = find_primes(r)
        results.append([r, count, duration])
        print(f"Completed range: {r}")
    
    with open('python_results.csv', 'w', newline='') as f:
        writer = csv.writer(f)
        writer.writerow(['Range', 'Primes Found', 'Duration'])
        writer.writerows(results)

if __name__ == "__main__":
    run_test()
